Clarify that the results of division operations on integer array data types resulting in floating-point outputs is implementation-defined (#362)
data-apis/array-api: spec/API_specification/elementwise_functions.md

Calculates the division for each element `x1_i` of the input array `x1` with the respective element `x2_i` of the input array `x2`.

```{note}

> If one or both of the input arrays have integer data types, the result is unspecified and thus implementation-dependent.

Reviewer: Don't know if this needs a note or not, but if we're doing this, we should make this say "floating-point" https://github.com/data-apis/array-api/pull/362/files#diff-c63388022ee724b37174b4191598444df80aacb366d1f63677da4d14784df654R566

Reviewer: Also for both divide and pow, if we do have a note would be clearer to have the note in the parameters section, so that it is right next to the part that says "numeric data types" or "floating-point data types".

Author: > Also for both divide and pow, if we do have a note would be clearer to have the note in the parameters section, so that it is right next to the part that says "numeric data types" or "floating-point data types".

Ah I agree that would make sense. For divide functions this seems tricky though, because the note also talks about the result dtype. Splitting the notes kinda loses the connectedness of the input and output.

Calculates the division for each element `x1_i` of the input array `x1` with the respective element `x2_i` of the input array `x2`.

```{note}

> If one or both of the input arrays have integer data types, the result is implementation-dependent. This is because type promotion between data type “kinds” (i.e. integer versus floating-point) is unspecified.

Reviewer 2: ```suggestion
If one or both of the input arrays have integer data types, the result is implementation-dependent, as type promotion between data type "kinds" (e.g., integer versus floating-point) is unspecified.
```

> Specification-compliant libraries may choose to raise an error or return an array containing the element-wise results. If an array is returned, it must have a floating-point data type.

Reviewer 3: Ditto.

Reviewer 2: Same as above.

> Specification-compliant libraries may choose to raise an error or return an array containing the element-wise results. If an array is returned, it must have a floating-point data type.

Reviewer 2: ```suggestion
Specification-compliant libraries may choose to raise an error or return an array containing the element-wise results. If an array is returned, the array must have a floating-point data type.
```

```{note}
If both `x1` and `x2` have integer data types, the result of `pow` when `x2_i` is negative (i.e., less than zero) is unspecified and thus implementation-dependent.

> If `x1` has an integer data type and `x2` has a floating-point data type, behavior is implementation-dependent. This is because type promotion between data type “kinds” (i.e. integer versus floating-point) is unspecified.

Reviewer 2: ```suggestion
If `x1` has an integer data type and `x2` has a floating-point data type, behavior is implementation-dependent, as type promotion between data type "kinds" (e.g., integer versus floating-point) is unspecified.
```
